src/mumble/MumbleAPI.cpp

```diff
@@ -0,0 +1,434 @@
+	if (!g.sh || g.sh->getConnectionID() != connection) {
+		return EC_CONNECTION_NOT_FOUND;
+	}
+
+	ClientUser *user = ClientUser::get(userID);
+
+	if (!user) {
+		return EC_USER_NOT_FOUND;
+	}
+
+	Channel *channel = Channel::get(channelID);
+
+	if (!channel) {
+		return EC_CHANNEL_NOT_FOUND;
+	}
+
+	// TODO: handle password-protected channels
+	// probably the easiest way to do so would be to silently add the given PW (if any) to the user's access tokens
+	Q_UNUSED(password);
+
+	if (channel != user->cChannel) {
+		// send move-request to the server only if the user is not in the channel already
+		g.sh->joinChannel(user->uiSession, channel->iId);
+	}
+
+	return STATUS_OK;
+}
+
+MumbleError_t PLUGIN_CALLING_CONVENTION findUserByName_v_1_0_0(MumbleConnection_t connection, const char *userName, MumbleUserID_t *userID) {
+	// Right now there can only be one connection managed by the current ServerHandler
+	if (!g.sh || g.sh->getConnectionID() != connection) {
+		return EC_CONNECTION_NOT_FOUND;
+	}
+
+	const QString qsUserName = QString::fromUtf8(userName);
+	
+	QReadLocker userLock(&ClientUser::c_qrwlUsers);
+
+	QHash<unsigned int, ClientUser*>::const_iterator it = ClientUser::c_qmUsers.constBegin();
+	while (it != ClientUser::c_qmUsers.constEnd()) {
+		if (it.value()->qsName == qsUserName) {
+			*userID = it.key();
+
+			return STATUS_OK;
+		}
+
+		it++;
+	}
+
+	return EC_USER_NOT_FOUND;
+}
+
+MumbleError_t PLUGIN_CALLING_CONVENTION findChannelByName_v_1_0_0(MumbleConnection_t connection, const char *channelName, MumbleChannelID_t *channelID) {
+	// Right now there can only be one connection managed by the current ServerHandler
+	if (!g.sh || g.sh->getConnectionID() != connection) {
+		return EC_CONNECTION_NOT_FOUND;
+	}
+
+	const QString qsChannelName = QString::fromUtf8(channelName);
+	
+	QReadLocker channelLock(&Channel::c_qrwlChannels);
+
+	QHash<int, Channel*>::const_iterator it = Channel::c_qhChannels.constBegin();
+	while (it != Channel::c_qhChannels.constEnd()) {
+		if (it.value()->qsName == qsChannelName) {
+			*channelID = it.key();
+
+			return STATUS_OK;
+		}
+
+		it++;
+	}
+
+	return EC_CHANNEL_NOT_FOUND;
+}
+
+MumbleError_t PLUGIN_CALLING_CONVENTION sendData_v_1_0_0(MumbleConnection_t connection, MumbleUserID_t *users, size_t userCount, const char *data,
+	size_t dataLength, const char *dataID) {
+	// Right now there can only be one connection managed by the current ServerHandler
+	if (!g.sh || g.sh->getConnectionID() != connection) {
+		return EC_CONNECTION_NOT_FOUND;
+	}
+
+	// TODO: Implement plugin-messages
+	Q_UNUSED(users);
+	Q_UNUSED(userCount);
+	Q_UNUSED(data);
+	Q_UNUSED(dataLength);
+	Q_UNUSED(dataID);
+
+	return EC_GENERIC_ERROR;
+}
+
+MumbleAPI getMumbleAPI(Version_t apiVersion) {
+	// Select the set of API functions for the requested API version
+	if (apiVersion.major == 1 && apiVersion.minor == 0 && apiVersion.patch == 0) {
+		// v1.0.0
+		return { freeMemory_v_1_0_0,
```

Comment:
is there really a need for 3 digit version number? Last number should just be hotfixes and no api changes, so there shouldn't be the need for a different function?